File: AdvancePython/Day-2 Constructors and Methods/Task-3-5.py
# Task 3:
# 5. Mutators:
class Car:
    def __init__(self, speed: int) -> None:
        self._speed: int = speed

    @property
    def speed(self) -> int:
        return self._speed

    @speed.setter  # the mutator
    def speed(self, value: int) -> None:
        assert value >= 0, "Speed cannot be negative"
        self._speed = value


my_car = Car(60)
my_car.speed = 80  # uses the setter
print(f"Car running at {my_car.speed} km/h.")

my_car.speed = -10  # checks the validation(mutator) logic and gives assertion error

File: node.js/Day-2/Task-3-5.js
// Task 3:
// 5. Mutators (Setters):

class Car {
    constructor(speed) {
        this._speed = speed
    }

    get speed() {
        return `Car speed is ${this._speed} km/h`;
    }
    
    set speed(curSpeed) {
        if(curSpeed < 0) {
            console.log("Speed can not be negative!");
        }
        this._speed = curSpeed;
    }
}

const car = new Car(60);
car.speed = 50;
console.log(car.speed);